tracer/src/Datadog.Trace/Headers/HeadersCollectionAdapter.cs

```diff
@@ -23,7 +23,7 @@ public IEnumerable<string> GetValues(string name)
         {
             if (_headers.TryGetValue(name, out var values))
             {
-                return values.ToArray();
+                return values;
```

Comment:
[`StringValues`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringvalues) implements `IEnumerable<string>`. I think a boxing operation is slightly better than the array allocation.

Reply:
And the boxing would go away in #2270 if we go that route.